### PR TITLE
added separate email mongo store

### PR DESCRIPTION
**Pull request by AndyHWChung**

### CODE REVIEW

**Comment by AndyHWChung** [comment]
https://jira.tools.tax.service.gov.uk/browse/HTS-413

File: app/uk/gov/hmrc/helptosavefrontend/controllers/RegisterController.scala [hunk 1]
```diff
@@ -65,25 +87,29 @@ class RegisterController @Inject()(val messagesApi: MessagesApi,
...
-                // start the process to enrol the user but don't worry about the result
-                enrolmentService.enrolUser(nino, userInfo.contactDetails.email).fold(
-                  e ⇒ logger.warn(s"Could not start process to enrol user $nino: $e"),
-                  _ ⇒ logger.info(s"Started process to enrol user $nino")
-                )
+          checkIfDoneEligibilityChecks { case (nsiUserInfo, confirmedEmail) ⇒
+            confirmedEmail.fold[Future[Result]](
+              SeeOther(routes.RegisterController.getConfirmDetailsPage().url)
+            ) { email ⇒
+              // TODO: plug in actual pages below
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavefrontend/controllers/RegisterController.scala]: remove this TODO, its confusing now as we create account just  after it

File: app/uk/gov/hmrc/helptosavefrontend/controllers/RegisterController.scala [hunk 2]
```diff
@@ -65,25 +87,29 @@ class RegisterController @Inject()(val messagesApi: MessagesApi,
...
+              helpToSaveService.createAccount(nsiUserInfo.updateEmail(email)).leftMap(submissionFailureToString).fold(
+                error ⇒ {
+                  // TODO: error or warning?
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavefrontend/controllers/RegisterController.scala]: we can remove this TODO too as i think its should be a error,

File: app/uk/gov/hmrc/helptosavefrontend/repo/EmailStore.scala [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    EitherT[Future,String,Unit](
+      doUpdate(email, nino)
+        .map{ _.fold[Either[String,Unit]](
+          Left("Could not update email mongo store"))(
+          _ ⇒ Right(()))
+        }
+        .recover{
+          case NonFatal(e) ⇒
+            Left(e.getMessage)
+        })
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavefrontend/repo/EmailStore.scala]: not necessary, but you can format above code

File: test/uk/gov/hmrc/helptosavefrontend/repo/MongoTestSupport.scala [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+import play.modules.reactivemongo.ReactiveMongoComponent
+import reactivemongo.api.DefaultDB
+import reactivemongo.bson.BSONObjectID
+import uk.gov.hmrc.mongo.{MongoConnector, ReactiveRepository}
+
+import scala.concurrent.Future
+
+trait MongoTestSupport[Data, R <: ReactiveRepository[Data,BSONObjectID]] { this: MockFactory ⇒
+
+  trait MockDBFunctions {
```
Comment (sureshhmrc) [inline test/uk/gov/hmrc/helptosavefrontend/repo/MongoTestSupport.scala]: if we search in gihub for `MongoTestSupport` we see how others are mocking mongo stuff they dont use `MockDBFunctions`

**Comment by sureshhmrc** [comment]
LGTM

**Pull request merged**